### PR TITLE
chore: Upgrading spring to 3.3.3 to resolve vulnerable dependencies

### PR DESCRIPTION
**Pull request by mohanarpit**
## Description
This PR updates Spring and it's associated dependencies to resolve vulnerable dependencies as flagged by Dependabot and Docker Scout.

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11011995792>
> Commit: 725c1a746c9679be56017ad49c44d17ade9cb8bd
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11011995792&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Git`
> Spec:
> <hr>Tue, 24 Sep 2024 11:03:23 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Updated the Spring Boot parent dependency to version `3.3.3`, enhancing application performance and stability.
	- Modified the default text source in configuration to pull from the entire `data` object, potentially improving data display.

- **Bug Fixes**
	- Corrected the URL format in the datasource configuration for proper API connectivity.
	- Removed specific version declarations for `spring-boot-starter-webflux` across multiple plugins, streamlining dependency management and reducing potential inconsistencies.

- **Chores**
	- Enhanced error handling in MongoDB plugin tests for more accurate simulation of error scenarios.

### CODE REVIEW

File: app/server/pom.xml [hunk 1]
```diff
@@ -48,8 +48,7 @@
         <snakeyaml.version>2.0</snakeyaml.version>
         <source.disabled>true</source.disabled>
         <spotless.version>2.36.0</spotless.version>
-        <spring-boot.version>3.0.9</spring-boot.version>
-        <testcontainers.version>1.19.3</testcontainers.version>
+        <testcontainers.version>1.20.1</testcontainers.version>
```
Comment (nidhi-nair) [inline app/server/pom.xml:51]: Guessing mongo failure is related to this?

File: app/server/appsmith-plugins/anthropicPlugin/pom.xml [hunk 2]
```diff
@@ -89,7 +89,6 @@
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-webflux</artifactId>
```
Comment (nidhi-nair) [inline app/server/appsmith-plugins/anthropicPlugin/pom.xml:91]: Interesting that we're asking for a test dependency here, we should be able to shade this from interfaces directly. Did you get any clues as to why the version was explicitly mentioned here before?

File: app/server/pom.xml (hunk 1 above)
Comment (mohanarpit) [reply app/server/pom.xml:51]: Yes. But this is required as per my understanding to fix a transitive dependency.

File: app/server/appsmith-server/pom.xml [hunk 3]
```diff
@@ -390,6 +384,12 @@
             <version>1.10.0</version>
         </dependency>
 
+        <dependency>
```
[the remark this answers is not in the document]
Comment (mohanarpit) [reply app/server/appsmith-server/pom.xml:387]: Yes. commons-codec was no longer included which was causing compile failure.

File: app/server/appsmith-plugins/anthropicPlugin/pom.xml (hunk 2 above)
Comment (mohanarpit) [reply app/server/appsmith-plugins/anthropicPlugin/pom.xml:91]: Good point. I don't remember why we did this.

File: app/server/appsmith-plugins/mongoPlugin/src/test/java/com/external/plugins/MongoPluginErrorsTest.java [hunk 4]
```diff
@@ -236,7 +239,23 @@ public void testGetStructureReadPermissionError() {
         when(mockConnection.getDatabase(any())).thenReturn(mockDatabase);
 
         MongoCommandException mockMongoCmdException = mock(MongoCommandException.class);
-        when(mockDatabase.listCollectionNames()).thenReturn(Mono.error(mockMongoCmdException));
+        // Mock the ListCollectionNamesPublisher
+        ListCollectionNamesPublisher mockPublisher = mock(ListCollectionNamesPublisher.class);
+
+        // Simulate an error when calling listCollectionNames
+        when(mockDatabase.listCollectionNames()).thenReturn(mockPublisher);
```
Comment (mohanarpit) [inline app/server/appsmith-plugins/mongoPlugin/src/test/java/com/external/plugins/MongoPluginErrorsTest.java:246]: @nidhi-nair Do we not have to make this similar change to `MongoPlugin` code itself?

Comment (nidhi-nair) [reply app/server/appsmith-plugins/mongoPlugin/src/test/java/com/external/plugins/MongoPluginErrorsTest.java:246]: In the plugin, we have a wrapper that converts the publisher to a flux already, [here](https://github.com/appsmithorg/appsmith/blob/cd3472ac1d7db9bfbd2bf54ae77cb69663159f00/app/server/appsmith-plugins/mongoPlugin/src/main/java/com/external/plugins/MongoPlugin.java#L960). That takes care of handling error scenarios. The mocking here is modifying the underlying subscription, which is automatically transferred to the reactive chain in the production code.

**Review by nidhi-nair — APPROVED** [review]

**Comment by nidhi-nair** [comment]
Do not merge, waiting on EE checks to pass. SSO is failing.

**Pull request merged**